src/include/storage/data_table.h

```diff
@@ -36,39 +37,36 @@ class DataTable {
   /**
    * Materializes a single tuple from the given slot, as visible at the timestamp.
    *
-   * @param txn_start_time the timestamp threshold that the returned projection should be visible at. In practice this
-   *                       will just be the start time of the caller transaction.
+   * @param txn the calling transaction
    * @param slot the tuple slot to read
    * @param out_buffer output buffer. The object should already contain projection list information. @see ProjectedRow.
    */
-  void Select(timestamp_t txn_start_time, TupleSlot slot, ProjectedRow *out_buffer) const;
+  void Select(transaction::TransactionContext *txn, TupleSlot slot, ProjectedRow *out_buffer) const;
 
   /**
    * Update the tuple according to the redo slot given, and update the version chain to link to the given
    * delta record. The delta record is populated with a before-image of the tuple in the process. Update will only
    * happen if there is no write-write conflict, otherwise, this is equivalent to a noop and false is returned,
    *
+   * @param txn the calling transaction
```

Comment:
Previously, we documented the undo record:
"the undo record to maintain and populate. It is expected that the projected row has the same structure as the redo, but the contents need not be filled beforehand, and will be populated with the before-image after this method returns."

We probably don't need the full details, but is it worth mentioning that the undo record is updated, or is it a no-brainer given the function name?

Reply:
Since the entire update behavior is now invisible to the caller (they don't need to know what happens in side transaction context), I think it's okay we leave that out.